tests/Http/HttpRequestTest.php

```diff
@@ -537,4 +537,22 @@ public function testCreateFromBase()
 
         $this->assertEquals($request->request->all(), $body);
     }
+
+    public function testJsonContent()
+    {
+        $body = [
+            'foo' => 'bar',
+            'baz' => ['qux'],
+        ];
+
+        $server = [
+            'CONTENT_TYPE' => 'application/json',
```

Comment:
Well technically, both will work, but the Symfony request will translate `HTTP_CONTENT_TYPE` to `CONTENT_TYPE`:

```php
/**
 * Creates a new request with values from PHP's super globals.
 *
 * @return Request A new request
 *
 * @api
 */
public static function createFromGlobals()
{
    // With the php's bug #66606, the php's built-in web server
    // stores the Content-Type and Content-Length header values in
    // HTTP_CONTENT_TYPE and HTTP_CONTENT_LENGTH fields.
    $server = $_SERVER;
    if ('cli-server' === php_sapi_name()) {
        if (array_key_exists('HTTP_CONTENT_LENGTH', $_SERVER)) {
            $server['CONTENT_LENGTH'] = $_SERVER['HTTP_CONTENT_LENGTH'];
        }
        if (array_key_exists('HTTP_CONTENT_TYPE', $_SERVER)) {
            $server['CONTENT_TYPE'] = $_SERVER['HTTP_CONTENT_TYPE'];
        }
    }
    $request = self::createRequestFromFactory($_GET, $_POST, array(), $_COOKIE, $_FILES, $server);
    if (0 === strpos($request->headers->get('CONTENT_TYPE'), 'application/x-www-form-urlencoded')
        && in_array(strtoupper($request->server->get('REQUEST_METHOD', 'GET')), array('PUT', 'DELETE', 'PATCH'))
    ) {
        parse_str($request->getContent(), $data);
        $request->request = new ParameterBag($data);
    }
    return $request;
}
```

In particular:

```php
if (array_key_exists('HTTP_CONTENT_TYPE', $_SERVER)) {
    $server['CONTENT_TYPE'] = $_SERVER['HTTP_CONTENT_TYPE'];
}
```